packages/graphql/src/services/questions.ts

```diff
@@ -415,3 +416,112 @@ export async function getFileUploadSas(
     fileName: blobName,
   }
 }
+
+export async function updateQuestionInstances(
+  { questionId }: { questionId: number },
+  ctx: ContextWithUser
+) {
+  // fetch the question and return null, if the question does not exist
+  const question = await ctx.prisma.question.findUnique({
+    where: {
+      id: questionId,
+    },
+    include: {
+      instances: {
+        include: {
+          sessionBlock: {
+            include: {
+              session: true,
+            },
+          },
+          stackElement: {
+            include: {
+              stack: {
+                include: {
+                  learningElement: {
+                    where: {
+                      status: DB.LearningElementStatus.DRAFT,
+                    },
+                  },
+                },
+              },
+            },
+          },
+          microSession: {
+            where: {
+              status: DB.MicroSessionStatus.DRAFT,
+            },
+          },
+        },
+      },
+    },
+  })
+
+  if (!question) {
+    return null
+  }
+
+  // get all instances and the corresponding element multipliers
+  const instanceMultipliers: Record<number, number> = {
+    ...question.instances.reduce((prev1, instance) => {
```

Comment:
A bit nitpicking, but I think `prev1`  is not a very intuitive variable name.. What would you think of renaming it to something like `acc` referring to the accumulator which gathers all the results from previous iterations?🤔